mozilla/blurts-server
# MNTOR-1664: Floating newsletter banner

# References: 
Jira: [MNTOR-1664](https://mozilla-hub.atlassian.net/browse/MNTOR-1664)
Figma: https://www.figma.com/file/unnNb6GYuNPnrIr1yS61Yh/Optin-Monster-Monitor-Banners?type=design&node-id=0-1&t=XoXWHvZIWiUdjrjf-0



# Description

This adds an optional floating banner. The banner should be only shown for certain pages and with a delay of `x` seconds.

# Screenshot

<img width="1271" alt="floating-banner-proposal-01" src="https://github.com/mozilla/blurts-server/assets/13835474/8bae5e9b-f01f-49a7-b520-2b293407e49c">


# How to test

1. Set the following `.env` variables:
```
FLOATING_BANNER_DELAY=5000
FLOATING_BANNER_LINK='https://www.mozilla.org/newsletter/security-and-privacy/'
FLOATING_BANNER_TYPE='newsletterSignupBanner'
FLOATING_BANNER_PAGES='/breaches,/user/breaches'
```
2. The Banner should be shown after a delay of `5000 ms` when navigating to `/breaches` or `/user/breaches`.
3. After following the sign-up link or dismissing the banner once a user should not see the banner again.

# Checklist (Definition of Done)
